lib/core/include/irods/irods_logger.hpp

```diff
@@ -484,14 +485,49 @@ namespace irods::experimental::log
             ///
             /// \since 4.3.0
             template <typename... Args>
-            auto operator()(fmt::format_string<Args...> _format, Args&&... _args) const -> void
+            void operator()(fmt::format_string<Args...> _format, Args&&... _args) const
+                // Clang 15 and later (and GCC 11 and later) consider this overload ambiguous
+                // with the std::invocable overload. This constraint sorts that out.
+#if __clang__ == 0 || __clang_major__ > 15
+                requires (sizeof...(Args) != 1 || !std::is_invocable_v<std::tuple_element_t<0, std::tuple<Args...>>>)
+#elif __clang_major__ == 15
+                // Clang 15 can't parse the above statement properly, so we have special overloads for 15.
+                // This overload handles the 1-arguemnt case.
+                requires (sizeof...(Args) == 0)
+#endif
             {
                 if (should_log()) {
                     const auto msg = {key_value{tag::log::message, fmt::format(_format, std::forward<Args>(_args)...)}};
                     log_message(std::begin(msg), std::end(msg));
                 }
             } // operator()
 
+#if __clang_major__ == 15
+            // This overload covers the 2-argument cases
+            template <typename Arg>
+            void operator()(fmt::format_string<Arg> _format, Arg&& _arg) const
+                requires (!std::is_invocable_v<Arg>)
+            {
+                if (should_log()) {
+                    const auto msg = {key_value{tag::log::message, fmt::format(_format, std::forward<Args>(_arg))}};
+                    log_message(std::begin(msg), std::end(msg));
+                }
+            }
```

Comment:
Let's add the `// operator()` to the ends of these functions for consistency.

Reply:
Done